src/fastoad/openmdao/variables.py

```diff
@@ -576,6 +576,15 @@ def from_problem(
                 if "." not in metadata["prom_name"]
             }
 
+            if get_promoted_names:
+                # Check connections
+                for name, metadata in inputs.copy().items():
```

Comment:
Simple comment, no change needed:
Generally speaking, I try to avoid copying items as much as I can.
I understand why you iterate on `inputs.copy()` instead of `inputs`, but I don't think it is good practice, because it makes you iterate twice on the dict (once for the copy, and then for your loop).
A better solution would probably be to iterate on the original dict and build a list of the variables that should be transferred from inputs to outputs, and then do the transfer by iterating this list (that will be far smaller that the dict).
Anyway, there is no performance issue in this part of the code, so we can live with that.

Reply:
Noted, I had thought about that...